### PR TITLE
Deprecate infinity values in data documents

### PR DESCRIPTION
**Pull request by noisersup**
# Description

This PR is a part of the #1151.

## Readiness checklist

* [x] I added tests for new functionality or bugfixes.
* [x] I ran `task all`, and it passed.
* [ ] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] I ensured that the title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/types/document_validation.go [hunk 1]
```diff
@@ -38,9 +39,9 @@ func (e *ValidationError) Error() string {
 // ValidateData checks if the document represents a valid "data document".
 // If the document is not valid it returns *ValidationError.
 func (d *Document) ValidateData() error {
-	// The following block should be used to checks that keys are valid.
-	// All further key related validation rules should be added here.
```
Comment (noisersup) [inline internal/types/document_validation.go:42]: @rumyantseva is there any background behind making this loop only for keys? Can I leave the change or should I make another loop for values?

File: internal/types/document.go [hunk 2]
```diff
@@ -93,6 +93,10 @@ func NewDocument(pairs ...any) (*Document, error) {
 		}
 	}
 
+	if err := doc.ValidateCmd(); err != nil {
+		return nil, fmt.Errorf("types.NewDocument: %w", err)
+	}
+
```
Comment (noisersup) [inline internal/types/document.go:99]: Not sure if my reasoning behind command documents is correct

File: internal/types/document_validation.go (hunk 1 above)
Comment (rumyantseva) [reply internal/types/document_validation.go:42]: @noisersup up to you. Please take a looks at https://github.com/FerretDB/FerretDB/pull/1293, there won't be separate way to store values soon, so ost likely, both keys and values will be in the same loop.

File: internal/types/document.go (hunk 2 above)
Comment (rumyantseva) [reply internal/types/document.go:99]: This is a good question! Not every document is a command document. I think `ValidateCmd` should be called from handlers. @AlekSi, please correct me if I'm wrong.

**Review by w84thesun — APPROVED** [review]
LGTM 👍

File: integration/commands_administration_test.go [hunk 3]
```diff
@@ -876,8 +876,6 @@ func TestCommandsAdministrationServerStatusMetrics(t *testing.T) {
 	} {
 		name, tc := name, tc
 		t.Run(name, func(t *testing.T) {
-			t.Parallel()
```
Comment (rumyantseva) [inline integration/commands_administration_test.go:879]: Why did you remove it?

File: internal/types/document_validation_test.go [hunk 4]
```diff
@@ -46,6 +47,14 @@ func TestDocumentValidateData(t *testing.T) {
 			doc:    must.NotFail(NewDocument("$v", "bar")),
 			reason: errors.New(`invalid key: "$v" (key must not contain $)`),
 		},
+		"Inf+": {
+			doc:    must.NotFail(NewDocument("v", math.Inf(1))),
+			reason: errors.New(`invalid value: +Inf (infinity values are not allowed)`),
```
Comment (rumyantseva) [inline internal/types/document_validation_test.go:52]: Let's also log the field that gave an invalid value (so, it'll be easier to debug and just more user-friendly in general).

**Review by rumyantseva — CHANGES_REQUESTED** [review]
Overal looks good to me.
I have a few minor questions, plus we need to document the difference in `diff.md`. 
Also, from my perspective, the tests you removed should be moved to `dance` (maybe you have already considered it).

File: integration/query_comparison_test.go [hunk 5]
```diff
@@ -588,19 +572,11 @@ func TestQueryComparisonGt(t *testing.T) {
...
-				"double", "double-big", "double-max", "double-positive-infinity", "double-smallest", "double-whole",
+				"double", "double-big", "double-max", "double-smallest", "double-whole",
 				"int32", "int32-max",
 				"int64", "int64-big", "int64-max",
 			},
 		},
-		"DoubleMax": {
-			value:       math.MaxFloat64,
-			expectedIDs: []any{"double-positive-infinity"},
-		},
```
Comment (chilagrow) [inline integration/query_comparison_test.go:599]: We don't need to remove this test case.

File: integration/commands_administration_test.go (hunk 3 above)
Comment (noisersup) [reply integration/commands_administration_test.go:879]: I believe that's the reason why pg tests fail sometimes. `serverStatus` is the command which is not related to specific collection, so running these test cases in parallel makes it possible to read values in "wrong" order

File: internal/types/document_validation_test.go (hunk 4 above)
Comment (noisersup) [reply internal/types/document_validation_test.go]: Does it look ok?

File: integration/query_comparison_test.go (hunk 5 above)
Comment (noisersup) [reply integration/query_comparison_test.go:599]: Nice catch!

**Review by chilagrow — APPROVED** [review]
:rocket:

File: integration/commands_administration_test.go (hunk 3 above)
Comment (AlekSi) [reply integration/commands_administration_test.go:879]: But what part of the test expects a particular order?

File: internal/types/document_validation_test.go (hunk 4 above)
Comment (rumyantseva) [reply internal/types/document_validation_test.go]: Yeah, that looks good to me.

File: integration/commands_administration_test.go (hunk 3 above)
Comment (noisersup) [reply integration/commands_administration_test.go:879]: Nevermind, now looking at tests it looks that they are order agnostic

**Review by AlekSi — APPROVED** [review]

**Review by rumyantseva — CHANGES_REQUESTED** [review]
I didn't get an answer to the question from my past review:
- when will we update diff.md
- when will we add dance tests

**Comment by AlekSi** [comment]
> when will we add dance tests

It is impossible to do that in this PR.

> when will we update diff.md

When we will be back working on this issue.

That PR is good enough otherwise. It does not close the issue.

**Review by rumyantseva — APPROVED** [review]
All the questions are answered, and the code looks good to me.

**Review by chilagrow — APPROVED** [review]
